Add RUM Shopify integration docs (#3076)
DataDog/integrations-extras: rum_shopify/README.md

Reviewer: claude has flagged that this file is missing a **Data Collected** section with Metrics / Service Checks / Events, which are required for new integrations. even if this information isn't collected, the headers should still exist and say something like, "The Shopify (RUM) integration does not include any metrics."

# Shopify Integration

## Overview

Instrument your Shopify store with Datadog Real User Monitoring using the Datadog RUM Shopify bundle. One bundle covers both surfaces of a Shopify store:

> - **Storefront pages** (`/`, `/products/*`, `/collections/*`, etc.) - added via a Theme Liquid snippet. Full RUM feature set: views, resources, long tasks, user interactions, errors, and Session Replay.

Reviewer: ```suggestion
- **Storefront pages** (`/`, `/products/*`, `/collections/*`, etc.) - added through a Theme Liquid snippet. Full RUM feature set: views, resources, long tasks, user interactions, errors, and Session Replay.
```

> - **Checkout pages** (`/checkouts/*`, `/checkout`) - added via a Custom Pixel. Shopify deprecated `checkout.liquid` in August 2024, so checkout pages cannot be reached by theme edits at all; a [Web Pixel][5] running in a sandboxed iframe is the only mechanism that can observe them.

Reviewer: ```suggestion
- **Checkout pages** (`/checkouts/*`, `/checkout`) - added through a Custom Pixel. Shopify deprecated `checkout.liquid` in August 2024, so checkout pages cannot be reached by theme edits at all; a [Web Pixel][5] running in a sandboxed iframe is the only mechanism that can observe them.
```

## Setup

### Prerequisites

Before you begin, gather the following Datadog RUM values:

- A Datadog RUM Application ID
- A Datadog RUM Client Token
- A Datadog site, such as `datadoghq.com`

You can get those values in Datadog under **Digital Experience > Real User Monitoring > Manage Applications > Set Up Manually**.

The snippets below are written for the US1 site. If your organization is on a different [Datadog site][4], substitute both of the following in every snippet:

- `us1` in the bundle URL, with your site's datacenter code (for example `eu1`, `us3`, `us5`, `ap1`).
- `<YOUR_DATADOG_SITE>`, with your site's domain (for example `datadoghq.eu`, `us3.datadoghq.com`).

You'll also need:

- Admin access to your Shopify store (**Settings** and **Online Store > Themes**).
- For the Custom Pixel path: access to **Settings > Customer events** in Shopify Admin.

### Installation and Configuration

#### Theme Liquid Setup

Use this path to cover storefront pages (product, collection, cart, home).

##### Steps

1. In Shopify Admin, go to **Online Store > Themes**.
2. On your live theme, click the kebab menu, then **Edit code**.
3. Open `layout/theme.liquid`.

> 4. Paste the following snippet right before the closing `</head>` tag, replacing the placeholder values with your Datadog RUM configuration:

Reviewer 2: I might be wrong but there could be a difference between adding it first or last? I think we recommend adding it first so it is executed before?

Author: Yeah, good spot! But at the same time I wound't give the recommendation to past it at the start of <head> sections as well before the <meta> tags. I believe we could use the text from the previous (legacy) [guide](https://docs.datadoghq.com/real_user_monitoring/guide/enable-rum-shopify-store/) and say: 
> Initialize the Browser RUM SDK by adding the SDK code snippet inside the \<head\> tag.

##### Steps

1. In Shopify Admin, go to **Settings > Customer events**.
2. Click **Add custom pixel**, give it a name, and open its code editor.
3. Paste the following snippet, replacing the placeholder values with your Datadog RUM configuration:

   ```javascript
   (function (h, o, u, n, d) {
     h = h[d] = h[d] || { q: [], onReady: function (c) { h.q.push(c) } }
     d = o.createElement(u); d.src = n
     d.crossOrigin = 'anonymous'
     n = o.getElementsByTagName(u)[0]; n.parentNode.insertBefore(d, n)
   })(
     window,
     document,
     'script',
     'https://www.datadoghq-browser-agent.com/us1/v7/datadog-rum-shopify.js',
     'DD_RUM'
   )

   DD_RUM.onReady(function () {
     DD_RUM.init({
       applicationId: '<YOUR_DATADOG_APPLICATION_ID>',
       clientToken: '<YOUR_DATADOG_CLIENT_TOKEN>',
       site: '<YOUR_DATADOG_SITE>',
       service: '<YOUR_SERVICE_NAME>',
       env: '<YOUR_ENV_NAME>',
       version: '1.0.0',
       sessionSampleRate: 100,
       shopifyAnalytics: analytics,
     })
   })
   ```

> 4. Save the pixel, then set its required consent category under **Settings** for the pixel (e.g. Analytics) to match your store's privacy configuration - the bundle does not add its own consent gate; it defers entirely to whatever the merchant declares in Shopify's Pixel Manager.

Reviewer: ```suggestion
4. Save the pixel, then set its required consent category under **Settings** for the pixel (for example, Analytics) to match your store's privacy configuration - the bundle does not add its own consent gate; it defers entirely to whatever the merchant declares in Shopify's Pixel Manager.
```

You can also emit and subscribe to [custom events][8] the same way.

##### Limitations

- **No Session Replay or Profiling.** The sandbox iframe has no real checkout DOM - recording or profiling it would capture nothing meaningful. Both are force-disabled regardless of what you pass to `init()`.
- **No Web Vitals, resource timing, or runtime JS errors.** `PerformanceObserver`, `fetch`/`XMLHttpRequest` instrumentation, and `window.onerror`/`unhandledrejection` all observe the sandbox iframe itself, not the checkout page's main thread - that boundary can't be crossed from a Custom Pixel.
- **Click actions are partial.** Only Shopify's basic `clicked` DOM event is available (no automatic DOM click listener, since there's no real DOM to listen on):
  - Emitted for inputs, links, and buttons - not for empty space or plain text.
  - Not emitted for payment card fields (card number, expiration, CVC, name on card) or the "Sign in" link.
  - Not emitted for a second click on a field that's already focused.
- **Advanced DOM Events are unavailable.** Shopify restricts that API to apps approved for it in the Partner Dashboard - a bar that doesn't apply to (and can't be met by) a Custom Pixel.

> - **Checkout-path detection is regex-based**, matching `/checkouts?/` with an optional two-letter locale prefix (e.g. `/en/checkout`). A store with a non-standard checkout URL structure won't currently have an override flag to adjust it.

Reviewer: ```suggestion
- **Checkout-path detection is regex-based**, matching `/checkouts?/` with an optional two-letter locale prefix (for example, `/en/checkout`). A store with a non-standard checkout URL structure won't currently have an override flag to adjust it.
```

### Validate the Installation

#### Storefront (Theme Liquid)

1. Open your storefront in a new browser session with developer tools open.
2. Confirm `datadog-rum-shopify.js` loads successfully (Network tab) and `window.DD_RUM` is defined (Console tab).
3. Navigate between a few storefront pages (home, a product, the cart).
4. In Datadog, open **Digital Experience > RUM Explorer**, filter by your configured `service` and `env`.
5. Confirm view events appear for each page you visited, and that a Session Replay recording is available for the session.

#### Checkout (Custom Pixel)

1. Place a test order (or go far enough into checkout to trigger `checkout_started`) on a storefront that has the Custom Pixel configured.
2. In the RUM Explorer, filter by the same `service`/`env` and look for views whose URL is under `/checkouts/`.
3. Confirm the view URL matches the real checkout URL (not a sandbox/iframe address) - this validates the `page_viewed`-driven `startView` binding is working.
4. Click a few checkout fields/buttons during the test run, then confirm corresponding click actions appear on the view.

> 5. If you added extra event subscriptions (see [Instrumenting extra Shopify events](#instrumenting-extra-shopify-events)), confirm their custom actions or user attributes (e.g. `checkout_completed`, the identified email from `setUser`) show up on the same view/session.

Reviewer: ```suggestion
5. If you added extra event subscriptions (see [Instrumenting extra Shopify events](#instrumenting-extra-shopify-events)), confirm their custom actions or user attributes (such as `checkout_completed`, the identified email from `setUser`) show up on the same view/session.
```

6. If nothing appears: check the pixel's consent category under **Settings > Customer events** - in opt-in (GDPR-style) regions, Shopify won't load the sandbox at all until the visitor consents, so no events fire until then.

## Troubleshooting

Need help? Contact [Datadog Support][3].

> [1]: https://shopify.dev/docs/api/web-pixels-api

Reviewer: this link is never referenced in the doc (also a duplicate) and should be removed. once this is deleted, please update the link numbering accordingly.

Author: It's referenced in the Overview section, but good catch on duplicates 👍

[2]: https://docs.datadoghq.com/help/

> [3]: https://docs.datadoghq.com/help/

Reviewer: this is a duplicate
